firecracker-control/local.go

```diff
@@ -78,7 +78,7 @@ func newLocal(ic *plugin.InitContext) (*local, error) {
 // CreateVM creates new Firecracker VM instance. It creates a runtime shim for the VM and the forwards
 // the CreateVM request to that shim. If there is already a VM created with the provided VMID, then
 // AlreadyExists is returned.
-func (s *local) CreateVM(requestCtx context.Context, req *proto.CreateVMRequest) (*empty.Empty, error) {
+func (s *local) CreateVM(requestCtx context.Context, req *proto.CreateVMRequest) (*proto.CreateVMResponse, error) {
```

Comment:
I like the idea of returning metadata from `CreateVM` but feel like if we are going to include the cgroup path in the `CreateVM` response it should also be included in the `GetVMInfoResponse` object.

That in turn makes me wonder if we should just have `CreateVM` return everything that's currently in `GetVMInfoResponse`, something like:
```
message VMInfo {
    string VMID = 1;
    uint32 ContextID = 2;
    string SocketPath = 3;
    string LogFifoPath = 4;
    string MetricsFifoPath = 5;
    string CgroupPath = 6;
}

message CreateVMResponse {
    VMInfo VMInfo = 1;
}

service Firecracker {
    rpc CreateVM(CreateVMRequest) returns (CreateVMResponse);

    rpc GetVMInfo(GetVMInfoRequest) returns (VMInfo);

    ...
}
```

Reply:
@sipsma - I like the idea, but I don't know if I like the example as much. I am on the fence on whether we should use a response object as another object in another response. I would be more in favor of something like this,

```
message GetVMInfoResponse {
    string VMID = 1;
    uint32 ContextID = 2;
    string SocketPath = 3;
    string LogFifoPath = 4;
    string MetricsFifoPath = 5;
    string CgroupPath = 6;
}

message CreateVMResponse {
    string VMID = 1;
    uint32 ContextID = 2;
    string SocketPath = 3;
    string LogFifoPath = 4;
    string MetricsFifoPath = 5;
    string CgroupPath = 6;
}

service Firecracker {
    rpc CreateVM(CreateVMRequest) returns (CreateVMResponse);

    rpc GetVMInfo(GetVMInfoRequest) returns (GetVMInfoResponse);

    ...
}
```

This makes the request and response names consistent and better for our users as they can easily predict what the response or request names are going to be. I don't mind the duplication here as there is a clear separation between the APIs. However, if that is something a lot of people disagree with I'd be okay with introducing the `VMInfo` structure, and have that returned in both as a field.

Reply:
Sure, I think that's fine too, the duplication is pretty minor maintenance overhead and would possibly be desirable anyways if we ever end up needing to diverge in the future.

Reply:
Awesome, fixed!